devito/passes/clusters/cse.py

```diff
@@ -270,8 +274,12 @@ def _toposort(exprs):
 
     def choose_element(queue, scheduled):
         # Try to honor temporary names as much as possible
-        first = sorted(queue, key=lambda i: str(i.lhs)).pop(0)
-        queue.remove(first)
+        tmps = [i for i in queue if isinstance(i.lhs, CTemp)]
+        if tmps:
+            first = sorted(tmps, key=lambda i: i.lhs.name).pop(0)
+            queue.remove(first)
+        else:
+            first = queue.pop()
```

Comment:
why not pop(0) here?

are you not potentially swapping the order of `Eq(u...)` and `Eq(v...)` otherwiswe?

Reply:
queue is a `deque` it's not "indexable", pop alwats remove the first element in the queue. `pop(0)` is actually invalid and will throw an error.

Reply:
oh, right, it's a deque...